guides/API Query Language.md

```diff
@@ -1,131 +1,197 @@
+* `$ne`: Matches all values that are not equal to a specified value.
+* `$gt`: Matches values that are greater than a specified value.
+* `$gte`: Matches values that are greater than or equal to a specified value.
+* `$lt`: Matches values that are less than a specified value.
+* `$lte`: Matches values that are less than or equal to a specified value.
+* `$in`: Matches any of the values specified in an array.
+* `$nin`: Matches none of the values specified in an array.
+* `$startsWith`: Matches strings that start with a specified value. Not case-sensitive.
+* `$isEmpty`: Matches strings or arrays that are empty or not empty,
+  depending on whether the specified operand is `true` or `false`.
+
+#### Logical operators
+
+* `$and`: Joins query clauses with a logical _AND_
+  and returns all items that match the conditions of both clauses.
+* `$or`: Joins query clauses with a logical _OR_
+  and returns all items that match the conditions of either clause.
+* `$not`: Inverts the effect of a query expression
+  and returns items that don't match the query expression.
+
+#### Element operators
+
+* `$exists`: Matches items that have the specified field.
+
+#### Array operators
+
+* `$hasAll`: Matches arrays that contain all elements specified in the query.
+* `$hasSome`: Matches arrays that contain at least one element specified in the query.
+
+### Sample queries
+
 In the following example, the compound query returns all entities where the status equals `"A"` and either `qty` is less than `30` or `item` starts with the character `p`:
-```javascript
+
+```json
 {
-  status: "A",
-  $or: [{qty: {$lt: 30}}, {item: {$begins: "p"}}]
+  "status": "A",
+  "$or": [
+    {
+      "qty": { "$lt": 30 }
+    },
+    {
+      "item": { "$begins": "p" }
+    }
+  ]
 }
 ```
 
 The following example queries entities where the field `tags` value is an array with exactly two elements, `"red"` and `"blank"`, in the specified order:
-```javascript
-{ tags: ["red", "blank"] }
+
+```json
+{
+  "tags": [ "red", "blank" ]
+}
 ```
 
 The following example queries for all entities where `tags` is an array that contains the string `"red"` as one of its elements, or that `tags` is the string `"red"`:
-```javascript
-{tags: "red" } 
+
+```json
+{
+  "tags": "red"
+}
 ```
 
 The following query matches entities that do not contain the `item` field, or where the `item` field has no value:
-```javascript
-{item: {$exists: false }} 
+
+```json
+{
+  "item": { "$exists": false }
+}
 ```
 
-### The Sort Section
+## The `sort` section
+
 The sort section is an array of field names and sort order. If the order is not specified, it will be sorted in ascending order:
-```javascript
-sort: [{"fieldName":"sortField1"},{"fieldName":"sortField2","order":"DESC"}]
+
+```json
+{
+  "sort": [
+    {
+      "fieldName": "sortField1"
+    },
+    {
+      "fieldName": "sortField2",
+      "order": "DESC"
+    }
+  ]
+}
 ```
 
-### The Paging Section
+## The `paging` section
```

Comment:
`Queryv2` with `cursorPaging` is now the more common api, I think it has to be added .
See here:
https://github.com/wix-private/p13n/blob/master/protos/query/src/main/proto/wix/common/query.proto#L69

The type of paging is *per api*, most support only one of them (offset or cursor).

Reply:
done